packit/schema.py

```diff
@@ -483,18 +489,48 @@ class PackageConfigSchema(CommonConfigSchema):
     """
 
     jobs = fields.Nested(JobConfigSchema, many=True)
+    packages = fields.Dict(
+        keys=fields.String(), values=fields.Nested(CommonConfigSchema())
+    )
 
     # list of deprecated keys and their replacement (new,old)
     deprecated_keys = (("upstream_package_name", "upstream_project_name"),)
 
     @pre_load
-    def ordered_preprocess(self, data, **kwargs):
-        data = self.rename_deprecated_keys(data, **kwargs)
+    def ordered_preprocess(self, data: dict, **_) -> dict:
+        """Rename deprecated keys, and set defaults for 'packages' and 'jobs'
+
+        Args:
+            data: configuration dictionary as loaded from packit.yaml
+
+        Returns:
+            Transformed configuration dictionary with defaults
+            for 'packages' and 'jobs' set.
+        """
+        # Create a deepcopy(), so that loading doesn't modify the
+        # dictionary received.
+        data = copy.deepcopy(data)
+        data = self.rename_deprecated_keys(data)
+        # Don't use 'setdefault' in this case, as we should expect
+        # downstream_package_name only if there is no 'packages' key.
+        if "packages" not in data:
+            package_name = data.pop("downstream_package_name")
+            paths = data.pop("paths", ["./"])
+            data["packages"] = {
+                package_name: {
+                    "downstream_package_name": package_name,
+                    "paths": paths,
+                }
+            }
         data.setdefault("jobs", get_default_jobs())
-        data = self.propagate_options_to_jobs(data, **kwargs)
+        # By this point, we expect both 'packages' and 'jobs' to be present
+        # in the config.
+        data = self.rearrange_packages(data)
+        data = self.rearrange_jobs(data)
+        logger.debug(f"Repo config after pre-loading:\n{json.dumps(data, indent=4)}")
```

Comment:
@lachmanfrantisek @lbarcziova Is this what you meant by logging the configuration after the inheritance work was done? Please give it a try with `packit -d srpm` to see what this produces. Isn't this a little but too much information?

Reply:
just tried it and it is true that with a more complex configuration, this can be really noisy...I cannot really decide what to do, it may be beneficial to see the resulting configuration, but it is true that the information in the resulting config is mostly duplicated

Reply:
I'll leave it like this for now. It can be removed if it becomes and issue for some reason.